test-files/test_gap_7577_generator_prototype_rooting.ts

```
@@ -0,0 +1,72 @@
+  return "pad-" + i + "-" + (i * 7919) + "-" + (i % 13);
+}
+
+// --- the chain, on a single instance ---
+const g0 = small(2);
+const p0 = Object.getPrototypeOf(g0);
+console.log("A proto is object:", typeof p0 === "object" && p0 !== null);
+console.log("A proto === g.prototype:", p0 === small.prototype);
+console.log("A proto !== instance:", (p0 as unknown) !== (g0 as unknown));
+
+// `g.prototype` identity is stable across reads.
+console.log("B stable:", small.prototype === small.prototype);
```

Biome (2.5.6)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)